File: src/main/kotlin/net/yttravelers/spotshare/entities/Hotel.kt
package net.yttravelers.spotshare.entities

import net.yttravelers.spotshare.entities.codes.HotelKindDiv
import javax.persistence.*

@Entity
@Table(name = "hotel")
class Hotel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var hotelId: Int? = null

    @Column(nullable = false)
    var hotelName: String = ""

    @ManyToOne
    @JoinColumns(
            JoinColumn(name="hotelKindDiv", referencedColumnName="division"),
            JoinColumn(name="hotelKindDivKey", referencedColumnName="divKey")
    )
    var hotelKindDiv: HotelKindDiv? = null

    var hotelSummary: String = ""

    var hotelAddress: String = ""

    var hotelUrl: String = ""

}

File: src/main/angular/src/app/shared/model/hotel.ts
/**
 * ホテル
 */
export interface IHotel {
  /** ホテルID */
  hotelId?: number;
  /** ホテル名称 */
  hotelName?: string;
  /** ホテル種類区分 */
  hotelKindDiv?: number;
  /** ホテル種類区分名称 */
  hotelKindDivName?: string;
  /** 概要 */
  hotelSummary?: string;
  /** 住所 */
  hotelAddress?: string;
  /** URL */
  hotelUrl?: string;
}
